src/Utilities/set_eval.jl

```diff
@@ -0,0 +1,178 @@
+
+function _correct_distance_to_set(d::Real)
+    return ifelse(d < 0, zero(d), d)
+end
+
+function _correct_distance_to_set(d::AbstractVector{T}) where {T <: Real}
+    return [ifelse(di < 0, zero(T), di) for di in d]
+end
+
+function distance_to_set(v, s)
+    length(v) != MOI.dimension(s) && throw(DimensionMismatch("Mismatch between v and s"))
```

Comment:
I would write `Mismatch between vector and set`

Reply:
`v` can also be a scalar, `Mismatch between value and set`?

Reply:
That would work